assert/assertions.go

```diff
@@ -491,12 +491,12 @@ func validateEqualArgs(expected, actual interface{}) error {
 	return nil
 }
 
-// Same asserts that two pointers reference the same object.
+// Same asserts that two arguments reference the same object.
```

Comment:
This is ambiguous for slices in this case:
```go
var (
	a   [8]int
	sa0 = a[:]
	sa3 = a[3:]
)
assert.Same(t, sa0, sa3)
```
sa0 and sa3 reference the same _object_ if you take object to mean _array_, but not if you mean _contiguous segment of an underlying array_. The language specification uses both terms at various points. We should make it clear that `Same` only considers slices to be the same if they reference the same _contiguous segment of an underlying array_.

Reply:
IMO reverse is surprising
```go
var (
	a   [8]int
	sa0 = a[:]
	sa1 = a[:1]
)
assert.Same(t, sa0, sa1) // passes
```